src/platform/bouffalolab/BL602/NetworkCommissioningDriver.cpp

```diff
@@ -208,27 +189,34 @@ void BLWiFiDriver::ConnectNetwork(ByteSpan networkId, ConnectCallback * callback
     }
 }
 
-CHIP_ERROR BLWiFiDriver::StartScanWiFiNetworks(ByteSpan ssid)
+void BLWiFiDriver::ScanNetworks(ByteSpan ssid, WiFiDriver::ScanCallback * callback)
 {
-    CHIP_ERROR err = CHIP_NO_ERROR;
+    int iret = 0;
 
-    if (!ssid.empty())
-    {
-        memset(WiFiSSIDStr, 0, sizeof(WiFiSSIDStr));
-        memcpy(WiFiSSIDStr, ssid.data(), ssid.size());
-        err       = (CHIP_ERROR) wifi_mgmr_scan_adv(NULL, NULL, NULL, 0, NULL, WiFiSSIDStr, 1, 0);
-        scan_type = 1;
-    }
-    else
-    {
-        err       = (CHIP_ERROR) wifi_mgmr_scan(NULL, NULL);
-        scan_type = 0;
-    }
-    if (err != CHIP_NO_ERROR)
+    if (callback != nullptr)
     {
-        return CHIP_ERROR_INTERNAL;
+        if (!ssid.empty())
+        {
+            memset(mScanSSID, 0, sizeof(mScanSSID));
+            memcpy(mScanSSID, ssid.data(), ssid.size());
+            iret      = wifi_mgmr_scan_adv(NULL, NULL, NULL, 0, NULL, mScanSSID, 1, 0);
```

Comment:
How does the callee know where the ssid ends?  In particular, what happens here if `ssid.size() == sizeof(mScanSSID)`?

Or worse yet if it's longer, of course...  Then you have a buffer overrun on the memcpy.  But let's assume `kMaxWiFiSSIDLength` really does mean that.  Though I don't know that it's enforced up the callstack here.

Reply:
ssid.size() should be less than or equal to kMaxWiFiSSIDLength.
If ssid.size() is greater than kMaxWiFiSSIDLength, there must be something wrong in call path.
To be safety, I add value check on ssid.size() before memcpy. If check failed, return error to caller.
```c
            if (ssid.size() <= sizeof(mScanSSID)) {
                memset(mScanSSID, 0, sizeof(mScanSSID));
                memcpy(mScanSSID, ssid.data(), ssid.size());
                iret      = wifi_mgmr_scan_adv(NULL, NULL, NULL, 0, NULL, mScanSSID, 1, 0);
                mScanType = 1;
            }
            else {
                iret = -1;
            }
```